### ISSUE TITLE
Initial setup falls on a loop if steamservice.so isn't found

### ISSUE THREAD

**Issue by lolotraverso** [dasisdormax/cs2-multiserver#3]
When running the initial setup, the script seems to execute the steam console client and throws an error when loading Steam API:
`Loading Steam API...dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory`
After that, it seems to continue with the attempt to log in, but even if it is successful as it is my case (I had previously logged in using the steam client) it returns ok, finishes that section, and then comes back to asking for the username again. Here is the full operation log (I omitted the username):

```
To install and update the CS2 game server, you need to log in to your
Steam account.

Please enter your Steam account's username (the one you log in with, not
your display name).

> Your steam username: (username)

We will now try to log you in with that username. Steam will probably ask
you for your password and Steam Guard code.

Redirecting stderr to '/home/steam/.steam/logs/stderr.txt'
Logging directory: '/home/steam/.steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation - version 1698778122
-- type 'quit' to exit --
Loading Steam API...dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
OK
Logging in user '(username) to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
quit


Please enter your Steam account's username (the one you log in with, not
your display name).
```

I guess the fact that it enters the loop might be a bug, but where does the script look for 'steamservice.so' so I can create a symlink?

Thank you!

**Comment by lolotraverso** [comment]
Just tested the steamcmd console again and it does present the same steam API failure, although steamcmd ignores that and shows the steam console prompt.

**Comment by dasisdormax** [comment]
SteamCMD showing that error is normal, it works fine without steamservice.so, and there is no way I know to add it.

There is a different issue here. Apparently, my script did not detect that you logged in successfully after SteamCMD exited. Can you check the file `~/Steam/config/config.vdf` and check if it has a section like this?

```
	"Accounts"
	{
		"(username)"
		{
			"SteamID"		"..."
		}
	}
```

**Comment by lolotraverso** [comment]
Hey,

Thanks for your time and effort :)

I don't have config.vdf located there, its under /home/steam/.steam/config/config.vdf, and it does contains that, and the account and SteamID are the right ones.
Might that be the reason why it doesn't detect it? if so, a symLink could do as a workaround?


Again, thank you!

**Comment by dasisdormax** [comment]
Thanks for your response. 

Putting a Symlink from Steam/config to .steam/config should do the trick for you.

I'm still investigating why the config file is in different locations. Maybe I need to modify the script to look in both places for the config file.

**Comment by dasisdormax** [comment]
Please check if the latest version on main (commit 92ca14a) fixes your issue.

I couldn't reproduce the issue on my end, so I'll just check for the file in both locations.

**Comment by dasisdormax** [comment]
I did another fix with aa7c3e4. Apparently, Steam uses yet another config location if the regular Steam client is installed on the system.
